openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import subprocess
+from uuid import uuid4
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class BlockDevice:
+    """BlockDevice provides functionality to map image files to block devices
+    and mount them.
+
+    Usage:
+        bd = BlockDevice('/folder/path_to_disk_image.dd')
+        mountpoints = bd.mount()
+        # Do the things you need to do :)
+        bd.destroy()
+    """
+
+    def __init__(self, image_path: str):
+        """Initialize BlockDevice class instance.
+
+        Args:
+            image_path (str): path to the image file to map and mount.
+        """
+        self.image_path = image_path
+        self.blkdevice = None
+        self.blkdeviceinfo = None
+        self.partitions = []
+        self.mountpoints = []
+        self.mountroot = "/mnt"
+
+        # Setup the loop device
+        self._losetup()
+
+        # Parse block device info
+        self._blkinfo()
+
+        # Parse partition information
+        self._parse_partitions()
+
+    def _losetup(self):
+        """Map image file to loopback device using losetup.
+
+        Returns: None
+        """
+        losetup_command = [
+            "sudo",
+            "losetup",
+            "--find",
+            "--partscan",
+            "--show",
+            self.image_path,
+        ]
+
+        process = subprocess.run(
+            losetup_command, capture_output=True, check=False, text=True
+        )
+        if process.returncode == 0:
+            self.blkdevice = process.stdout.strip()
+        else:
+            raise RuntimeError(f"Error: {process.stderr} {process.stdout}")
+
+        return None
+
+    def _blkinfo(self):
```

Comment:
To make it easier to test this method, you could have it return the json.loads() output instead, and the caller can assign it to self.blkdeficeinfo instead. Otherwise, this always returns none which is less nice to test. WDYT?